src/java.base/share/classes/java/lang/classfile/attribute/LoadableDescriptorsAttribute.java

```diff
@@ -43,44 +44,28 @@
  *  deletion without notice.</b>
  */
 @PreviewFeature(feature = PreviewFeature.Feature.CLASSFILE_API)
-public sealed interface PreloadAttribute
-        extends Attribute<PreloadAttribute>, ClassElement
-        permits BoundAttribute.BoundPreloadAttribute, UnboundAttribute.UnboundPreloadAttribute {
+public sealed interface LoadableDescriptorsAttribute
+        extends Attribute<LoadableDescriptorsAttribute>, ClassElement
+        permits BoundAttribute.BoundLoadableDescriptorsAttribute, UnboundAttribute.UnboundLoadableDescriptorsAttribute {
 
     /**
-     * {@return the list of preload classes}
+     * {@return the list of loadable descriptors}
      */
-    List<ClassEntry> preloads();
+    List<Utf8Entry> loadableDescriptors();
 
     /**
-     * {@return a {@code Preload} attribute}
-     * @param preloads the preload classes
+     * {@return a {@code LoadableDescriptors} attribute}
+     * @param loadableDescriptors the loadable descriptors
      */
-    static PreloadAttribute of(List<ClassEntry> preloads) {
-        return new UnboundAttribute.UnboundPreloadAttribute(preloads);
+    static LoadableDescriptorsAttribute of(List<Utf8Entry> loadableDescriptors) {
+        return new UnboundAttribute.UnboundLoadableDescriptorsAttribute(loadableDescriptors);
     }
 
     /**
-     * {@return a {@code Preload} attribute}
-     * @param preloads the preload classes
+     * {@return a {@code LoadableDescriptors} attribute}
+     * @param loadableDescriptors the loadable descriptors
      */
-    static PreloadAttribute of(ClassEntry... preloads) {
-        return of(List.of(preloads));
-    }
-
-    /**
-     * {@return a {@code Preload} attribute}
-     * @param preloads the preload classes
-     */
-    static PreloadAttribute ofSymbols(List<ClassDesc> preloads) {
-        return of(Util.entryList(preloads));
-    }
-
-    /**
-     * {@return a {@code Preload} attribute}
-     * @param preloads the preload classes
-     */
-    static PreloadAttribute ofSymbols(ClassDesc... preloads) {
-        return ofSymbols(Arrays.asList(preloads));
+    static LoadableDescriptorsAttribute of(Utf8Entry... loadableDescriptors) {
```

Comment:
I recommend another set of factory:
```java
static LoadableDescriptorsAttribute ofSymbols(ClassDesc... loadableDescriptors);
static LoadableDescriptorsAttribute ofSymbols(List<ClassDesc> loadableDescriptors);
```

Reply:
I don't see the benefit tbh

Reply:
we can always add it if necessary in the future